dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -226,4 +264,180 @@ private void checkModelPluralForm(String apiCategory, String model) {
 			throw new RepositoryNotFoundException(apiCategory, model);
 		}
 	}
+	
+	@RequestMapping(method = RequestMethod.GET, value="/search")
+	ResourceSupport listSearchMethods(@PathVariable String apiCategory, @PathVariable String model) {
+		ResourceSupport root = new ResourceSupport();
+		DSpaceRestRepository repository = utils.getResourceRepository(apiCategory, model);
+		boolean searchEnabled = false;
+		for (Method method : repository.getClass().getMethods()) {
+			SearchRestMethod ann = method.getAnnotation(SearchRestMethod.class);
+			if (ann != null) {
+				String name = ann.name();
+				if (name.isEmpty()) {
+					name = method.getName();
+				}
+				Link link = linkTo(this.getClass(), apiCategory, model).slash("search").slash(name).withRel(name);
+				root.add(link);
+				searchEnabled = true;
+			}
+		}
+		if (!searchEnabled) {
+			throw new RepositorySearchNotFoundException(model);
+		}
+		return root;
+	}
+	
+	
+	@RequestMapping(method = RequestMethod.GET, value="/search/{searchMethod}")
+	@SuppressWarnings("unchecked")
+	<T extends RestModel> PagedResources<DSpaceResource<T>> executeSearchMethods(@PathVariable String apiCategory, 
+			@PathVariable String model, @PathVariable String searchMethod, Pageable pageable, Sort sort, PagedResourcesAssembler assembler, 
+			@RequestParam MultiValueMap<String, Object> parameters) throws IllegalAccessException, IllegalArgumentException, InvocationTargetException {
+			
+		Link link = linkTo(this.getClass(), apiCategory, model).slash("search").slash(searchMethod).withSelfRel();
+		DSpaceRestRepository repository = utils.getResourceRepository(apiCategory, model);
+		Page<DSpaceResource<T>> resources = null;
+		boolean searchEnabled = false;
+		boolean searchMethodFound = false;
+		for (Method method : repository.getClass().getMethods()) {
```

Comment:
This for-loop could be moved to a `Method findSearchMethod(DSpaceRestRepository repository, String searchMethod)` in the reflection helper class.

Reply:
ok